docs/structured-streaming-kafka-integration.md

```diff
@@ -499,20 +522,28 @@ StreamingQuery ds = df
   .option("topic", "topic1")
   .start();
 
-// Write key-value data from a DataFrame to Kafka using a topic specified in the data
+// Write key-value data from a DataFrame to Kafka using the topic specified in the data
 StreamingQuery ds = df
   .selectExpr("topic", "CAST(key AS STRING)", "CAST(value AS STRING)")
   .writeStream()
   .format("kafka")
   .option("kafka.bootstrap.servers", "host1:port1,host2:port2")
   .start();
 
+// Write key-value data from a DataFrame to Kafka using the topic specified in the path option
+StreamingQuery ds = df
+  .selectExpr("CAST(key AS STRING)", "CAST(value AS STRING)")
```

Comment:
nit: Since you added it I'd change it to `.select($"key" cast "string", $"value" cast "string")` even if it's the only place in the doc. It's simply more type-safe.

Reply:
would it be better to change it in all places in this doc file if we anyway touch it?

Reply:
With the latest change don't fully understand the direction. Now half of the doc uses `selectExpr ` and the other part `select`. Still questioning the need, what is the benefit? If the `selectExpr` executed and properly formed not more unstable than the simple `select` in the doc right?

Reply:
select is used for scala, as we cannot use same syntax for Java and Python ($), but we can use 
`select(col("key").cast("string"))` for all three languages if we really want it to be typesafe and looks the same way.

Reply:
I see the intention now, thanks for the explanation. I personally would leave as it is but not telling  the direction is wrong. Let's decide by a committer who potentially picks this up.

Reply:
@jaceklaskowski what do you think: should we revert the changes here or use `select(col("key").cast("string"))` in all places to provide type safety?